### PR TITLE
bug(server): replicate scripts in stable state

### LINKED ISSUE #1040: Scripts are not transferred in stable sync
```py



SCRIPT_TEMPLATE = "return {}"

@dfly_args({"proactor_threads": 2})
async def test_script_transfer(df_local_factory):
    master = df_local_factory.create(port=BASE_PORT)
    replica = df_local_factory.create(port=BASE_PORT+1)

    df_local_factory.start_all([master, replica])

    c_master = aioredis.Redis(port=master.port)
    c_replica = aioredis.Redis(port=replica.port)

    # Load some scripts into master ahead
    scripts = []
    for i in range(0, 10):
        sha = await c_master.script_load(SCRIPT_TEMPLATE.format(i))
        scripts.append(sha)

    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
    await wait_available_async(c_replica)
    
    # transfer in stable state
    for i in range(10, 20):
        sha = await c_master.script_load(SCRIPT_TEMPLATE.format(i))
        scripts.append(sha)
    
    check_all_replicas_finished([c_replica], c_master)
    await c_replica.execute_command("REPLICAOF NO ONE")
    
    for i, sha in enumerate(scripts):
        print(i) # fails from 10 and on
        assert await c_replica.evalsha(sha, 0) == i
```

### PR DESCRIPTION
**Pull request by adiholden**
Fixes #1040 

The solution is this PR for replicating scripts:
Script is non key command and is not part of the transaction framework which is how we replicate commands.
Therefor I created a new command mask, if this mask is set I use the transaction framework using shard 0 to replicate the command

### CODE REVIEW

File: src/server/transaction.cc [hunk 1]
```diff
@@ -929,6 +943,12 @@ bool Transaction::ScheduleUniqueShard(EngineShard* shard) {
   DCHECK(shard_data_.size() == 1u || multi_->mode == NON_ATOMIC);
   DCHECK_NE(unique_shard_id_, kInvalidSid);
 
+  // callback with no args that needs to be scheduled can run ooo, no keys need to be locked.
+  if (empty_args_) {
+    RunQuickie(shard);
+    return true;
+  }
+
```
Comment (dranikpg) [inline src/server/transaction.cc:951]: Not sure we need this at all, because it doesn't have any keys either way and the lock check will pass.

Comment (dranikpg) [reply src/server/transaction.cc:951]: The other question is whether we should respect global locks - and it looks like we should.

For example, in-between switching journal subscribers from full to stable sync, we issue a global tx to block all journal events. Now this event might get simply lost for some replica if its fired during this window

**Review by dranikpg — APPROVED** [review]

File: src/server/transaction.cc (hunk 1 above)
Comment (adiholden) [reply src/server/transaction.cc:951]: There are many CHECK fails on this flow this is why I added this, but true we do need to respect the gloabal locks, I will fix this

**Review by dranikpg — APPROVED** [review]
Patching `GetLockArgs` to return empty args for NO_KEY_JOURNAL entries would have been enough (+ remove empty check in CheckLock), so it wouldn't change or complicate the code flow inside transaction logic

**Comment by adiholden** [comment]
@dranikpg what you suggest indeed makes the transaction code more simple but at the same time we have less checks in our code which makes it more prone to errors, f.e if we do a change in the logic that initialises the args and this leads to a bug that we are not locking some of them. IMHO the former solution the code is more complicated but more safe. Any way I did the changes, you can take a look and decide what you think is better

**Review by dranikpg — COMMENTED** [review]
I agree, but simpler code is easier to maintain and less prone to errors. Acquire is used only withing transaction itself, so if we don't want to loose the DCHECKs, we can just move them to tx as `DCHECK(!keys.empty() || cid & no_key_journal)`

If you go with this version, please fix the bug

**Review by dranikpg — APPROVED** [review]

**Review by romange — APPROVED** [review]

**Pull request merged**